openedx_learning/core/contents/api.py

```diff
@@ -80,69 +40,122 @@ def get_or_create_media_type_id(mime_type: str) -> int:
         suffix = ""
 
     main_type, sub_type = base.split("/")
-    mt, _created = MediaType.objects.get_or_create(
+    media_type, _created = MediaType.objects.get_or_create(
         type=main_type,
         sub_type=sub_type,
         suffix=suffix,
     )
 
-    return mt.id
+    return media_type
+
+
+def get_content(content_id: int, /) -> Content:
+    """
+    Get a single Content object by its ID.
+
+    Content is always attached to something when it's created, like to a
+    ComponentVersion. That means the "right" way to access a Content is almost
+    always going to be via those relations and not via this function. But I
+    include this function anyway because it's tiny to write and it's better than
+    someone using a get_or_create_* function when they really just want to get.
+    """
+    return Content.objects.get(id=content_id)
 
 
-def get_or_create_raw_content(
+def get_or_create_text_content(
     learning_package_id: int,
     /,
-    data_bytes: bytes,
-    mime_type: str,
+    media_type_id: int,
+    text: str,
     created: datetime,
-    hash_digest: str | None = None,
-) -> tuple[RawContent, bool]:
+    create_file: bool = False,
+) -> Content:
     """
-    Get the RawContent in the given learning package with the specified data,
-    or create it if it doesn't exist.
+    Get or create a Content entry with text data stored in the database.
+
+    Use this when you want to create relatively small chunks of text that need
+    to be accessed quickly, especially if you're pulling back multiple rows at
+    once. For example, this is the function to call when storing OLX for a
+    component XBlock like a ProblemBlock.
+
+    This function will *always* create a text entry in the database. In addition
+    to this, if you specify ``create_file=True``, it will also save a copy of
+    that text data to the file storage backend. This is useful if we want to let
+    that file be downloadable by browsers in the LMS at some point.
+
+    If you want to create a large text file, or want to create a text file that
+    doesn't need to be stored in the database, call ``create_file_content``
+    instead of this function.
+
+    The `hash_digest` is included as an optional argument because a very common
+    pattern is going to be "I have this data, let's see if a Content already
+    exists for it."
```

Comment:
```suggestion
```

Reply:
<sigh> Thank you for catching this. I went back and forth on it a few times before killing it off again.